File: java/src/main/java/ai/zentinelle/model/HeartbeatResult.java
package ai.zentinelle.model;

/**
 * Result of a heartbeat.
 */
public class HeartbeatResult {

    private final boolean acknowledged;
    private final boolean configChanged;
    private final int nextHeartbeatSeconds;

    private HeartbeatResult(Builder builder) {
        this.acknowledged = builder.acknowledged;
        this.configChanged = builder.configChanged;
        this.nextHeartbeatSeconds = builder.nextHeartbeatSeconds;
    }

    public boolean isAcknowledged() { return acknowledged; }
    public boolean isConfigChanged() { return configChanged; }
    public int getNextHeartbeatSeconds() { return nextHeartbeatSeconds; }

    public static Builder builder() { return new Builder(); }

    public static class Builder {
        private boolean acknowledged = true;
        private boolean configChanged;
        private int nextHeartbeatSeconds = 60;

        public Builder acknowledged(boolean acknowledged) { this.acknowledged = acknowledged; return this; }
        public Builder configChanged(boolean configChanged) { this.configChanged = configChanged; return this; }
        public Builder nextHeartbeatSeconds(int nextHeartbeatSeconds) {
            this.nextHeartbeatSeconds = nextHeartbeatSeconds;
            return this;
        }

        public HeartbeatResult build() { return new HeartbeatResult(this); }
    }
}

File: csharp/src/Zentinelle/Models/HeartbeatResult.cs
using System.Text.Json.Serialization;

namespace Zentinelle.Models;

/// <summary>
/// Result of a heartbeat request.
/// </summary>
public class HeartbeatResult
{
    /// <summary>
    /// Whether the heartbeat was accepted.
    /// </summary>
    [JsonPropertyName("acknowledged")]
    public bool Acknowledged { get; set; } = true;

    /// <summary>
    /// Explicit config-change signal from the service.
    /// </summary>
    [JsonPropertyName("config_changed")]
    public bool ConfigChanged { get; set; }

    /// <summary>
    /// Legacy drift signal used by some deployments.
    /// </summary>
    [JsonPropertyName("drift_detected")]
    public bool DriftDetected { get; set; }

    /// <summary>
    /// Legacy sync-required signal used by some deployments.
    /// </summary>
    [JsonPropertyName("sync_required")]
    public bool SyncRequired { get; set; }

    /// <summary>
    /// Suggested next heartbeat interval.
    /// </summary>
    [JsonPropertyName("next_heartbeat_seconds")]
    public int NextHeartbeatSeconds { get; set; } = 60;

    /// <summary>
    /// Returns true when any config-refresh signal is present.
    /// </summary>
    public bool HasConfigChangeSignal => ConfigChanged || DriftDetected || SyncRequired;
}
